beacon/goclient/goclient.go

```diff
@@ -141,19 +142,40 @@ var _ NodeClientProvider = (*GoClient)(nil)
 
 // GoClient implementing Beacon struct
 type GoClient struct {
-	log                  *zap.Logger
-	ctx                  context.Context
-	network              beaconprotocol.Network
-	client               Client
-	nodeVersion          string
-	nodeClient           NodeClient
-	gasLimit             uint64
-	operatorDataStore    operatordatastore.OperatorDataStore
+	log         *zap.Logger
+	ctx         context.Context
+	network     beaconprotocol.Network
+	client      Client
+	nodeVersion string
+	nodeClient  NodeClient
+	gasLimit    uint64
+
+	operatorDataStore operatordatastore.OperatorDataStore
+
 	registrationMu       sync.Mutex
 	registrationLastSlot phase0.Slot
 	registrationCache    map[phase0.BLSPubKey]*api.VersionedSignedValidatorRegistration
-	commonTimeout        time.Duration
-	longTimeout          time.Duration
+
+	// attestationReqMuPool helps us prevent the sending of multiple attestation data requests
+	// for the same slot number (to avoid doing unnecessary work).
+	// It's a pool of mutexes (not a single mutex) to allow for some parallelism when requests
+	// targeting different slots are made.
+	attestationReqMuPool [32]sync.Mutex
```

Comment:
The size of this `attestationReqMuPool` is pretty arbitrary (the more requests we send the higher this value should be), using just 1 mutex for all the requests is probably not enough, 32 seems reasonable here

Reply:
32 looks great, but theoretically, we might want to run a custom eth network in kurtosis in the future for tests/simulations, so it would be great to avoid hardcoding to current network constants for the new code. What do think about using a slice and initializing its length to `SlotsPerEpoch()`?

Reply:
Changed.